src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,304 @@
+package duke;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.EmptyDescriptionException;
+import duke.exceptions.InvalidKeywordException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+
+import java.util.ArrayList;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class Duke {
+
+    public static final String HORIZONTAL_LINE = "____________________________________________________________";
+    private static ArrayList<Task> tasks = new ArrayList<>();
+    private static int taskCount = 0;
+    private static String userInputLine;
+    private static String command;  // one question: when to make those variable static?
+    private static String description;
+    private static String taskName;
+    private static String taskTime;
+
+    public static void main(String[] args) throws IOException {
+        //auto load
+        File autoSaveFile = new File("data/duke.txt");
+        checkFileExistence(autoSaveFile);
+
+        Scanner in = new Scanner(System.in);
+        boolean isQuit = false;
+
+        //when the program starts
+        showLogo();
+        greet();
+
+        while(!isQuit) {
+            processInput(in);
+
+            switch (command.toUpperCase()) {
+            case "LIST":
+                listTasks();
+                break;
+            case "DONE":
+                if(checkDescription() && checkTaskNum()) {
+                    doneTask(Integer.parseInt(description));
+                }
+                break;
+            case "TODO":
+                if(checkDescription()) {
+                    addTask(new ToDo(description));
+                }
+                break;
+            case "DEADLINE":
+                if(checkDescription() && checkDeadline()) {
```

Comment:
It may be possible to further apply SLAP here and abstract the details of the checks